nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/HealthCheckService.java

```diff
@@ -230,8 +231,8 @@ public synchronized void bootstrapNotifierList() {
     }
 
     private void executeTasks() {
-        for (String registeredTaskKey : registeredTasks.keySet()) {
-            HealthCheckTask registeredTask = registeredTasks.get(registeredTaskKey);
+        for (Entry<String, HealthCheckTask> entry : registeredTasks.entrySet()) {
```

Comment:
This might even be a case where we could just iterate values directly. Otherwise there is a key replacement missing.